### PR TITLE
refactor(storage): Optimized `v1LegacyLayout` key creation.

### PR DESCRIPTION
**Pull request by alesforz**
This PR refactors the `v1LegacyLayout` key creation to improve its speed and reduce its memory usage.
Follow up to #4484.

---

#### PR checklist

- [x] Tests written/updated
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Comment by alesforz** [comment]
Benchmarks:
```
BenchmarkKeys/StringConcat-11               116908806                9.999 ns/op           0 B/op          0 allocs/op
BenchmarkKeys/Preallocate-11                   302796864                3.968 ns/op           0 B/op          0 allocs/op
```
`StringConcat` is what's in `main` now. `Preallocate` is the refactoring in this PR.

**Review by melekes — APPROVED** [review]
👍

File: light/store/db/db_key_layout.go [hunk 1]
```diff
@@ -21,7 +21,26 @@ type v1LegacyLayout struct{}
 
 // LBKey implements LightStoreKeyLayout.
 func (v1LegacyLayout) LBKey(height int64, prefix string) []byte {
-	return []byte(fmt.Sprintf("lb/%s/%020d", prefix, height))
+	const (
```
[the remark this answers is not in the document]
Comment (melekes) [reply light/store/db/db_key_layout.go:24]: yes

[the remark this answers is not in the document]
Comment (alesforz) [reply light/store/db/db_key_layout.go:24]: I ran some benchmarks, and they are the same using a global `const`. The length of the string doesn't affect the performance; I think the go compiler does some optimizations on `const` even if they are declared inside a function.

Benchmark with long `string` keys:
```
BenchmarkLongKeys/StringConcat-11    50991807    23.33 ns/op     8 B/op    1 allocs/op
BenchmarkLongKeys/Prealloc-11        137356957    8.736 ns/op    0 B/op    0 allocs/op
BenchmarkLongKeys/GlobalConst-11     137596965    8.730 ns/op    0 B/op    0 allocs/op
```

Benchmark with short `string` keys:
```
BenchmarkShortKey/Sprintf-11         24446098   48.82 ns/op    24 B/op    2 allocs/op
BenchmarkShortKey/Prealloc-11       161066708    7.446 ns/op    0 B/op    0 allocs/op
BenchmarkShortKey/GlobalConst-11    161569386    7.456 ns/op    0 B/op    0 allocs/op
```

Given these results, I propose to leave the `const` declaration inside the functions to not pollute the global namespace of the pkg 😄

**Pull request merged**